src/components/Movie.jsx

```diff
@@ -3,22 +3,23 @@ import starredSlice from '../data/starredSlice'
 import watchLaterSlice from '../data/watchLaterSlice'
 import placeholder from '../assets/not-found-500X750.jpeg'
 
-const Movie = ({ movie, viewTrailer, closeCard }) => {
+const Movie = ({ movie, viewTrailer }) => {
 
-    const state = useSelector((state) => state)
-    const { starred, watchLater } = state
+    const {starredMovies} = useSelector((state) => state.starred)
+    const  {watchLaterMovies} = useSelector((state) => state.watchLater)
     const { starMovie, unstarMovie } = starredSlice.actions
     const { addToWatchLater, removeFromWatchLater } = watchLaterSlice.actions
 
     const dispatch = useDispatch()
 
     const myClickHandler = (e) => {
-        if (!e) var e = window.event
-        e.cancelBubble = true
-        if (e.stopPropagation) e.stopPropagation()
+        e.stopPropagation()
         e.target.parentElement.parentElement.classList.remove('opened')
     }
 
+    const isStarred = starredMovies.some(_movie => _movie.id === movie.id);
+    const isWatchLater = watchLaterMovies.some(_movie => _movie.id === movie.id);
+
```

Comment:
Created variables in order to

1. Increased readability
2. Check ```starred``` & ```watched``` with the help of one array method.